src/migration/1730710262320-PostDeletedStatsTrigger.ts

```diff
@@ -0,0 +1,46 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class PostDeletedStatsTrigger1730710262320
+  implements MigrationInterface
+{
+  name = 'PostDeletedStatsTrigger1730710262320';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
+      CREATE OR REPLACE FUNCTION update_source_stats_on_delete()
+      RETURNS TRIGGER AS $$
+      BEGIN
+        UPDATE  source
+        SET     flags = flags || jsonb_build_object(
+                          'totalPosts',
+                          GREATEST(0, COALESCE(CAST(flags->>'totalPosts' AS INTEGER) , 0) - 1),
+                          'totalUpvotes',
+                          GREATEST(0, COALESCE(CAST(flags->>'totalUpvotes' AS INTEGER), 0) - NEW.upvotes),
+                          'totalViews',
+                          GREATEST(0, COALESCE(CAST(flags->>'totalViews' AS INTEGER), 0) - NEW.views)
+                        )
+        WHERE id = NEW."sourceId";
+        RETURN NEW;
+      END;
+      $$ LANGUAGE plpgsql;
+    `);
+    await queryRunner.query(
+      `
+        CREATE TRIGGER update_source_stats_on_delete
+        AFTER UPDATE ON "post"
+        FOR EACH ROW
+        WHEN (NEW.deleted <> OLD.deleted AND NEW.deleted IS TRUE)
```

Comment:
While this will remove counts once on soft delete the current triggers will still increment because they are missing `deleted` check?

Reply:
good point. it's an unlikely edge case but let me fix that

Reply:
fixed https://github.com/dailydotdev/daily-api/pull/2385/commits/ab591c987d22a1980a66949d99eb062e62fc5c67